compilation error in mitata.hpp

## detail

`fibonacci.cpp` won't be compiled.

## step to reproduce
```sh
cd example
g++ -std=c++20 fibonacci.cpp
```

<details>

<summary>
error output
</summary>

```
In file included from fibonacci.cpp:1:
../src/mitata.hpp:15:30: error: ‘uint8_t’ does not name a type
   15 |   typedef int8_t i8; typedef uint8_t u8;
      |                              ^~~~~~~
../src/mitata.hpp:12:1: note: ‘uint8_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   11 | #include <algorithm>
  +++ |+#include <cstdint>
   12 | 
../src/mitata.hpp:16:32: error: ‘uint16_t’ does not name a type
   16 |   typedef int16_t i16; typedef uint16_t u16;
      |                                ^~~~~~~~
../src/mitata.hpp:16:32: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/mitata.hpp:17:32: error: ‘uint32_t’ does not name a type
   17 |   typedef int32_t i32; typedef uint32_t u32;
      |                                ^~~~~~~~
../src/mitata.hpp:17:32: note: ‘uint32_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/mitata.hpp:18:32: error: ‘uint64_t’ does not name a type
   18 |   typedef int64_t i64; typedef uint64_t u64;
      |                                ^~~~~~~~
../src/mitata.hpp:18:32: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/mitata.hpp:23:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
   23 |       u64 ticks;
      |       ^~~
      |       i64
../src/mitata.hpp:38:36: error: ‘std::chrono’ has not been declared
   38 |     static const auto epoch = std::chrono::high_resolution_clock::now();
      |                                    ^~~~~~
../src/mitata.hpp:41:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
   41 |       u64 min_samples = k_min_samples;
      |       ^~~
      |       i64
../src/mitata.hpp:42:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
   42 |       u64 max_samples = k_max_samples;
      |       ^~~
      |       i64
../src/mitata.hpp:43:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
   43 |       u64 batch_unroll = k_batch_unroll;
      |       ^~~
      |       i64
../src/mitata.hpp:45:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
   45 |       u64 batch_samples = k_batch_samples;
      |       ^~~
      |       i64
../src/mitata.hpp:46:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
   46 |       u64 warmup_samples = k_warmup_samples;
      |       ^~~
      |       i64
../src/mitata.hpp:49:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
   49 |       u64 samples_threshold = k_samples_threshold;
      |       ^~~
      |       i64
../src/mitata.hpp: In function ‘const mitata::f64 mitata::lib::now()’:
../src/mitata.hpp:53:19: error: ‘std::chrono’ has not been declared
   53 |       return std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::high_resolution_clock::now() - epoch).count();
      |                   ^~~~~~
../src/mitata.hpp:53:46: error: ‘std::chrono’ has not been declared
   53 |       return std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::high_resolution_clock::now() - epoch).count();
      |                                              ^~~~~~
../src/mitata.hpp:53:72: error: ‘std::chrono’ has not been declared
   53 |       return std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::high_resolution_clock::now() - epoch).count();
      |                                                                        ^~~~~~
../src/mitata.hpp: In function ‘const mitata::lib::k_stats mitata::lib::fn(void (*)(), k_options)’:
../src/mitata.hpp:65:37: error: ‘const struct mitata::lib::k_options’ has no member named ‘warmup_samples’
   65 |           for (auto o = 0; o < opts.warmup_samples; o++) {
      |                                     ^~~~~~~~~~~~~~
../src/mitata.hpp:76:35: error: ‘const struct mitata::lib::k_options’ has no member named ‘max_samples’
   76 |         for (auto _ = 0; _ < opts.max_samples; _++) {
      |                                   ^~~~~~~~~~~
../src/mitata.hpp:77:25: error: ‘const struct mitata::lib::k_options’ has no member named ‘min_samples’
   77 |           if (_ >= opts.min_samples && t >= opts.min_cpu_time) break;
      |                         ^~~~~~~~~~~
../src/mitata.hpp:92:35: error: ‘const struct mitata::lib::k_options’ has no member named ‘max_samples’
   92 |         for (auto _ = 0; _ < opts.max_samples; _++) {
      |                                   ^~~~~~~~~~~
../src/mitata.hpp:93:25: error: ‘const struct mitata::lib::k_options’ has no member named ‘min_samples’
   93 |           if (_ >= opts.min_samples && t >= opts.min_cpu_time) break;
      |                         ^~~~~~~~~~~
../src/mitata.hpp:97:38: error: ‘const struct mitata::lib::k_options’ has no member named ‘batch_samples’
   97 |           for (auto o = 0; o < (opts.batch_samples / opts.batch_unroll); o++) {
      |                                      ^~~~~~~~~~~~~
../src/mitata.hpp:97:59: error: ‘const struct mitata::lib::k_options’ has no member named ‘batch_unroll’
   97 |           for (auto o = 0; o < (opts.batch_samples / opts.batch_unroll); o++) {
      |                                                           ^~~~~~~~~~~~
../src/mitata.hpp:98:39: error: ‘const struct mitata::lib::k_options’ has no member named ‘batch_unroll’
   98 |             for (auto u = 0; u < opts.batch_unroll; u++) fn();
      |                                       ^~~~~~~~~~~~
../src/mitata.hpp:105:47: error: ‘const struct mitata::lib::k_options’ has no member named ‘batch_samples’
  105 |           stats.samples.push_back(diff / opts.batch_samples);
      |                                               ^~~~~~~~~~~~~
../src/mitata.hpp:110:39: error: ‘const struct mitata::lib::k_options’ has no member named ‘samples_threshold’; did you mean ‘batch_threshold’?
  110 |       if (stats.samples.size() > opts.samples_threshold) stats.samples = std::vector<f64>(stats.samples.begin() + 2, stats.samples.end() - 2);
      |                                       ^~~~~~~~~~~~~~~~~
      |                                       batch_threshold
../src/mitata.hpp:119:13: error: ‘struct mitata::lib::k_stats’ has no member named ‘ticks’
  119 |       stats.ticks = stats.samples.size() * (!batch ? 1 : opts.batch_samples);
      |             ^~~~~
../src/mitata.hpp:119:63: error: ‘const struct mitata::lib::k_options’ has no member named ‘batch_samples’
  119 |       stats.ticks = stats.samples.size() * (!batch ? 1 : opts.batch_samples);
      |                                                               ^~~~~~~~~~~~~
../src/mitata.hpp: At global scope:
../src/mitata.hpp:151:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
  151 |       u64 avg;
      |       ^~~
      |       i64
../src/mitata.hpp:152:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
  152 |       u64 peak;
      |       ^~~
      |       i64
../src/mitata.hpp:153:7: error: ‘u64’ does not name a type; did you mean ‘i64’?
  153 |       u64 outliers;
      |       ^~~
      |       i64
../src/mitata.hpp:155:19: error: ‘u64’ was not declared in this scope; did you mean ‘i64’?
  155 |       std::vector<u64> bins;
      |                   ^~~
      |                   i64
../src/mitata.hpp:155:22: error: template argument 1 is invalid
  155 |       std::vector<u64> bins;
      |                      ^
../src/mitata.hpp:155:22: error: template argument 2 is invalid
../src/mitata.hpp:174:49: error: ‘u64’ has not been declared
  174 |     inline const std::string str(std::string s, u64 len = 3) {
      |                                                 ^~~
../src/mitata.hpp:179:51: error: ‘u64’ has not been declared
  179 |     inline const std::string pad_e(std::string s, u64 len, char c = ' ') {
      |                                                   ^~~
../src/mitata.hpp:186:51: error: ‘u64’ has not been declared
  186 |     inline const std::string pad_s(std::string s, u64 len, char c = ' ') {
      |                                                   ^~~
../src/mitata.hpp:205:50: error: ‘u64’ has not been declared
  205 |     inline const k_bins bins(lib::k_stats stats, u64 size = 6, f64 percentile = 1) {
      |                                                  ^~~
../src/mitata.hpp: In function ‘const mitata::fmt::k_bins mitata::fmt::bins(mitata::lib::k_stats, int, mitata::f64)’:
../src/mitata.hpp:206:7: error: ‘u64’ was not declared in this scope; did you mean ‘i64’?
  206 |       u64 poffset = percentile * (stats.samples.size() - 1);
      |       ^~~
      |       i64
../src/mitata.hpp:210:37: error: ‘poffset’ was not declared in this scope; did you mean ‘off_t’?
  210 |       f64 max = .0 != stats.samples[poffset] ? stats.samples[poffset] : (.0 == stats.max ? 1.0 : stats.max);
      |                                     ^~~~~~~
      |                                     off_t
../src/mitata.hpp:213:34: error: template argument 2 is invalid
  213 |       auto bins = std::vector<u64>(size, 0);
      |                                  ^
../src/mitata.hpp:217:53: error: ‘round’ is not a member of ‘std’
  217 |       for (auto o = 0; o <= poffset; o++) bins[std::round((stats.samples[o] - min) / step)]++;
      |                                                     ^~~~~
../src/mitata.hpp:224:30: error: expected ‘)’ before ‘std’
  224 |         .avg = clamp(0, (u64)std::round((stats.avg - min) / step), size - 1),
      |                     ~        ^~~
      |                              )
../src/mitata.hpp:225:7: error: could not convert ‘{min, max, step, bins, steps, <expression error>, <expression error>, <expression error>}’ from ‘<brace-enclosed initializer list>’ to ‘const mitata::fmt::k_bins’
  225 |       };
      |       ^
      |       |
      |       <brace-enclosed initializer list>
../src/mitata.hpp: At global scope:
../src/mitata.hpp:228:70: error: ‘u64’ has not been declared
  228 |     inline const std::string barplot(std::map<std::string, f64> map, u64 legend = 8, u64 width = 14, bool colors = true) {
      |                                                                      ^~~
../src/mitata.hpp:228:86: error: ‘u64’ has not been declared
  228 |     inline const std::string barplot(std::map<std::string, f64> map, u64 legend = 8, u64 width = 14, bool colors = true) {
      |                                                                                      ^~~
../src/mitata.hpp: In function ‘const std::string mitata::fmt::barplot(std::map<std::__cxx11::basic_string<char>, double>, int, int, bool)’:
../src/mitata.hpp:240:9: error: ‘u64’ was not declared in this scope; did you mean ‘i64’?
  240 |         u64 offset = 1 + std::round((value - min) / step);
      |         ^~~
      |         i64
../src/mitata.hpp:244:30: error: ‘offset’ was not declared in this scope; did you mean ‘off_t’?
  244 |         for (auto o = 0; o < offset; o++) barplot += "■"; if (colors) barplot += fmt::colors::reset;
      |                              ^~~~~~
      |                              off_t
../src/mitata.hpp: At global scope:
../src/mitata.hpp:256:66: error: ‘u64’ has not been declared
  256 |     inline const std::vector<std::string> histogram(k_bins bins, u64 height = 2, bool colors = true) {
      |                                                                  ^~~
../src/mitata.hpp: In function ‘const std::vector<std::__cxx11::basic_string<char> > mitata::fmt::histogram(k_bins, int, bool)’:
../src/mitata.hpp:260:102: error: request for member ‘size’ in ‘bins.mitata::fmt::k_bins::bins’, which is of non-class type ‘int’
  260 |       auto canvas = std::vector<std::vector<std::string>>(height, std::vector<std::string>(bins.bins.size(), " "));
      |                                                                                                      ^~~~
../src/mitata.hpp:262:7: error: ‘u64’ was not declared in this scope; did you mean ‘i64’?
  262 |       u64 avg = bins.avg;
      |       ^~~
      |       i64
../src/mitata.hpp:263:10: error: expected ‘;’ before ‘peak’
  263 |       u64 peak = bins.peak;
      |          ^~~~~
      |          ;
../src/mitata.hpp:264:56: error: ‘peak’ was not declared in this scope
  264 |       f64 scale = (f64)(height * symbols.size() - 1) / peak;
      |                                                        ^~~~
../src/mitata.hpp:266:38: error: request for member ‘size’ in ‘bins.mitata::fmt::k_bins::bins’, which is of non-class type ‘int’
  266 |       for (auto o = 0; o < bins.bins.size(); o++) {
      |                                      ^~~~
../src/mitata.hpp:267:27: error: invalid types ‘int[int]’ for array subscript
  267 |         auto b = bins.bins[o];
      |                           ^
../src/mitata.hpp:268:23: error: ‘round’ is not a member of ‘std’
  268 |         auto s = std::round(b * scale);
      |                       ^~~~~
../src/mitata.hpp:279:18: error: ‘avg’ was not declared in this scope
  279 |         if (0 != avg) {
      |                  ^~~
../src/mitata.hpp:285:24: error: ‘avg’ was not declared in this scope
  285 |         l += canvas[h][avg]; if (colors) l += fmt::colors::reset;
      |                        ^~~
../src/mitata.hpp:287:31: error: request for member ‘size’ in ‘bins.mitata::fmt::k_bins::bins’, which is of non-class type ‘int’
  287 |         if (avg != (bins.bins.size() - 1)) {
      |                               ^~~~
../src/mitata.hpp:289:48: error: request for member ‘size’ in ‘bins.mitata::fmt::k_bins::bins’, which is of non-class type ‘int’
  289 |           for (auto o = 1 + avg; o < bins.bins.size(); o++) l += canvas[h][o]; if (colors) l += fmt::colors::reset;
      |                                                ^~~~
../src/mitata.hpp: At global scope:
../src/mitata.hpp:298:79: error: ‘u64’ has not been declared
  298 |     inline const std::string boxplot(std::map<std::string, lib::k_stats> map, u64 legend = 8, u64 width = 14, bool colors = true) {
      |                                                                               ^~~
../src/mitata.hpp:298:95: error: ‘u64’ has not been declared
  298 |     inline const std::string boxplot(std::map<std::string, lib::k_stats> map, u64 legend = 8, u64 width = 14, bool colors = true) {
      |                                                                                               ^~~
../src/mitata.hpp: In function ‘const std::string mitata::fmt::boxplot(std::map<std::__cxx11::basic_string<char>, mitata::lib::k_stats>, int, int, bool)’:
../src/mitata.hpp:318:9: error: ‘u64’ was not declared in this scope; did you mean ‘i64’?
  318 |         u64 min_offset = std::round((min - tmin) / step);
      |         ^~~
      |         i64
../src/mitata.hpp:319:12: error: expected ‘;’ before ‘max_offset’
  319 |         u64 max_offset = std::round((max - tmin) / step);
      |            ^~~~~~~~~~~
      |            ;
../src/mitata.hpp:320:12: error: expected ‘;’ before ‘avg_offset’
  320 |         u64 avg_offset = std::round((avg - tmin) / step);
      |            ^~~~~~~~~~~
      |            ;
../src/mitata.hpp:321:12: error: expected ‘;’ before ‘p25_offset’
  321 |         u64 p25_offset = std::round((p25 - tmin) / step);
      |            ^~~~~~~~~~~
      |            ;
../src/mitata.hpp:322:12: error: expected ‘;’ before ‘p75_offset’
  322 |         u64 p75_offset = std::round((p75 - tmin) / step);
      |            ^~~~~~~~~~~
      |            ;
../src/mitata.hpp:324:47: error: ‘max_offset’ was not declared in this scope
  324 |         auto u = std::vector<std::string>(1 + max_offset, " ");
      |                                               ^~~~~~~~~~
../src/mitata.hpp:328:13: error: ‘min_offset’ was not declared in this scope
  328 |         if (min_offset < p25_offset) {
      |             ^~~~~~~~~~
../src/mitata.hpp:328:26: error: ‘p25_offset’ was not declared in this scope
  328 |         if (min_offset < p25_offset) {
      |                          ^~~~~~~~~~
../src/mitata.hpp:335:13: error: ‘p25_offset’ was not declared in this scope
  335 |         if (p25_offset < avg_offset) {
      |             ^~~~~~~~~~
../src/mitata.hpp:335:26: error: ‘avg_offset’ was not declared in this scope
  335 |         if (p25_offset < avg_offset) {
      |                          ^~~~~~~~~~
../src/mitata.hpp:339:21: error: ‘min_offset’ was not declared in this scope
  339 |           auto ms = min_offset == p25_offset ? "│" : "┤";
      |                     ^~~~~~~~~~
../src/mitata.hpp:344:11: error: ‘avg_offset’ was not declared in this scope
  344 |         u[avg_offset] = (!colors ? "┬" : (std::string(fmt::colors::yellow) + "┬" + fmt::colors::reset));
      |           ^~~~~~~~~~
../src/mitata.hpp:348:13: error: ‘p75_offset’ was not declared in this scope
  348 |         if (p75_offset > avg_offset) {
      |             ^~~~~~~~~~
../src/mitata.hpp:357:26: error: ‘p75_offset’ was not declared in this scope
  357 |         if (max_offset > p75_offset) {
      |                          ^~~~~~~~~~
../src/mitata.hpp:390:35: error: ‘floor’ is not a member of ‘std’
  390 |       boxplot += std::string(std::floor(gap), ' ') + ' ';
      |                                   ^~~~~
../src/mitata.hpp:395:41: error: ‘ceil’ is not a member of ‘std’
  395 |       boxplot += ' ' + std::string(std::ceil(gap), ' ');
      |                                         ^~~~
../src/mitata.hpp: In member function ‘std::map<std::__cxx11::basic_string<char>, mitata::lib::k_stats> mitata::runner::run(mitata::k_run)’:
../src/mitata.hpp:492:46: error: ‘struct mitata::lib::k_stats’ has no member named ‘ticks’
  492 |           std::cout << "\"ticks\": " << noop.ticks << "," << std::endl;
      |                                              ^~~~~
../src/mitata.hpp:522:47: error: ‘struct mitata::lib::k_stats’ has no member named ‘ticks’
  522 |               std::cout << "\"ticks\": " << s.ticks << "," << std::endl;
      |                                               ^~~~~
../src/mitata.hpp:693:26: error: ‘round’ is not a member of ‘std’
  693 |                   ? std::round(1 / c.avg * b.avg * 100) / 100
      |                          ^~~~~
../src/mitata.hpp:694:26: error: ‘round’ is not a member of ‘std’
  694 |                   : std::round(1 / b.avg * c.avg * 100) / 100;
      |                          ^~~~~
```

</details>

--- a/src/mitata.hpp
+++ b/src/mitata.hpp
@@ -220,11 +220,11 @@ namespace mitata {
       for (auto o = 0; o <= poffset; o++) bins[std::round((stats.samples[o] - min) / step)]++;
 
       return {
-        .avg = clamp(0, (u64)std::round((stats.avg - min) / step), size - 1),
-        .peak = *std::max_element(bins.begin(), bins.end()),
-        .outliers = stats.samples.size() - 1 - poffset,
         .min = min, .max = max,
         .step = step, .bins = bins, .steps = steps,
+        .outliers = stats.samples.size() - 1 - poffset,
+        .peak = *std::max_element(bins.begin(), bins.end()),
+        .avg = clamp(0, (u64)std::round((stats.avg - min) / step), size - 1),
       };
     }
 

--- a/src/mitata.hpp
+++ b/src/mitata.hpp
@@ -220,11 +220,11 @@ namespace mitata {
       for (auto o = 0; o <= poffset; o++) bins[std::round((stats.samples[o] - min) / step)]++;
 
       return {
+        .avg = clamp(0, (u64)std::round((stats.avg - min) / step), size - 1),
+        .peak = *std::max_element(bins.begin(), bins.end()),
+        .outliers = stats.samples.size() - 1 - poffset,
         .min = min, .max = max,
         .step = step, .bins = bins, .steps = steps,
-        .outliers = stats.samples.size() - 1 - poffset,
-        .peak = *std::max_element(bins.begin(), bins.end()),
-        .avg = clamp(0, (u64)std::round((stats.avg - min) / step), size - 1),
       };
     }
 

--- a/src/mitata.hpp
+++ b/src/mitata.hpp
@@ -12,6 +12,7 @@
 #include <numeric>
 #include <iostream>
 #include <algorithm>
+#include <functional>
 
 namespace mitata {
   typedef float f32; typedef double f64;

--- a/src/mitata.hpp
+++ b/src/mitata.hpp
@@ -3,8 +3,11 @@
 
 #include <map>
 #include <regex>
+#include <cmath>
+#include <chrono>
 #include <string>
 #include <vector>
+#include <cstdint>
 #include <sstream>
 #include <numeric>
 #include <iostream>

--- a/src/mitata.hpp
+++ b/src/mitata.hpp
@@ -15,10 +15,10 @@
 
 namespace mitata {
   typedef float f32; typedef double f64;
-  typedef int8_t i8; typedef uint8_t u8;
-  typedef int16_t i16; typedef uint16_t u16;
-  typedef int32_t i32; typedef uint32_t u32;
-  typedef int64_t i64; typedef uint64_t u64;
+  typedef std::int8_t i8; typedef std::uint8_t u8;
+  typedef std::int16_t i16; typedef std::uint16_t u16;
+  typedef std::int32_t i32; typedef std::uint32_t u32;
+  typedef std::int64_t i64; typedef std::uint64_t u64;
 
   namespace lib {
     struct k_stats {

--- a/src/mitata.hpp
+++ b/src/mitata.hpp
@@ -19,7 +19,6 @@ namespace mitata {
   typedef int16_t i16; typedef uint16_t u16;
   typedef int32_t i32; typedef uint32_t u32;
   typedef int64_t i64; typedef uint64_t u64;
-  typedef __int128 i128; typedef unsigned __int128 u128;
 
   namespace lib {
     struct k_stats {

